lib/Basic/Platform.cpp

```diff
@@ -431,147 +431,139 @@ llvm::Triple swift::getUnversionedTriple(const llvm::Triple &triple) {
-      } else {
-        return floorForArchitecture(llvm::VersionTuple(5, 2));
-      }
-    } else if (Major <= 14) {
-      if (Minor <= 4)
-        return floorForArchitecture(llvm::VersionTuple(5, 3));
-
-      return floorForArchitecture(llvm::VersionTuple(5, 4));
-    } else if (Major <= 15) {
-      if (Minor <= 3)
-        return floorForArchitecture(llvm::VersionTuple(5, 5));
-      return floorForArchitecture(llvm::VersionTuple(5, 6));
-    } else if (Major <= 16) {
-      if (Minor <= 3)
-        return floorForArchitecture(llvm::VersionTuple(5, 7));
-      return floorForArchitecture(llvm::VersionTuple(5, 8));
-    } else if (Major <= 17) {
-      if (Minor <= 3)
-        return floorForArchitecture(llvm::VersionTuple(5, 9));
-      return floorForArchitecture(llvm::VersionTuple(5, 10));
-    }
+
+    // iOS releases predate the stable ABI, so use Swift 5.0 as our base.
+    auto baseRelease = llvm::VersionTuple(5, 0);
+
+    // arm64 simulators and macCatalyst were introduced in iOS 14.0/tvOS 14.0,
+    // which included Swift 5.3.
+    if (Triple.isAArch64() &&
+        (Triple.isSimulatorEnvironment() ||
+         Triple.isMacCatalystEnvironment()))
+      baseRelease = llvm::VersionTuple(5, 3);
+
+    // iOS first got arm64e support in 12.0, which did not yet support the
+    // Swift stable ABI, so it does not provide a useful version bump.
+
+    return findSwiftRuntimeVersion(Platform::iOS, OSVersion, baseRelease);
+
   } else if (Triple.isWatchOS()) {
     llvm::VersionTuple OSVersion = Triple.getWatchOSVersion();
-    unsigned Major = OSVersion.getMajor();
-    unsigned Minor = OSVersion.getMinor().value_or(0);
-
-    auto floorFor64bits = [&Triple](llvm::VersionTuple v) {
-      if (!Triple.isArch64Bit()) return v;
-      // 64-bit watchOS was introduced with Swift 5.3
-      return MAX(v, llvm::VersionTuple(5, 3));
-    };
-
-    if (Major <= 5) {
-      return floorFor64bits(llvm::VersionTuple(5, 0));
-    } else if (Major <= 6) {
-      if (Minor <= 1) {
-        return floorFor64bits(llvm::VersionTuple(5, 1));
-      } else {
-        return floorFor64bits(llvm::VersionTuple(5, 2));
-      }
-    } else if (Major <= 7) {
-      if (Minor <= 3)
-        return floorFor64bits(llvm::VersionTuple(5, 3));
-
-      return floorFor64bits(llvm::VersionTuple(5, 4));
-    } else if (Major <= 8) {
-      if (Minor <= 4)
-        return floorFor64bits(llvm::VersionTuple(5, 5));
-      return floorFor64bits(llvm::VersionTuple(5, 6));
-    } else if (Major <= 9) {
-      if (Minor <= 3)
-        return floorFor64bits(llvm::VersionTuple(5, 7));
-      return floorFor64bits(llvm::VersionTuple(5, 8));
-    } else if (Major <= 10) {
-      if (Minor <= 3)
-        return floorFor64bits(llvm::VersionTuple(5, 9));
-      return floorFor64bits(llvm::VersionTuple(5, 10));
-    }
-  }
-  else if (Triple.isXROS()) {
+
+    // tvOS releases predate the stable ABI, so use Swift 5.0 as our base.
```

Comment:
Shouldn't this be watchOS?
```suggestion
    // watchOS releases predate the stable ABI, so use Swift 5.0 as our base.
```

Reply:
Yes, absolutely. Thanks for catching the slip-up.